media/client/ipc/source/MediaPipelineIpc.cpp

```diff
@@ -200,8 +201,30 @@ bool MediaPipelineIpc::attachSource(MediaSourceType type, const std::string &cap
     firebolt::rialto::AttachSourceRequest request;
 
     request.set_session_id(m_sessionId);
-    request.set_media_type(convertProtoMediaSourceType(type));
-    request.set_caps(caps);
+    auto mediaType = source.getType();
+    request.set_media_type(convertProtoMediaSourceType(mediaType));
+    request.set_mime_type(source.getMimeType());
+    if (SegmentAlignment::UNDEFINED != source.getSegmentAlignment())
+    {
+        request.set_segment_alignment(convertSegmentAlignment(source.getSegmentAlignment()));
+    }
+    AudioConfig audioConfig;
+    if (source.getAudioConfig(audioConfig))
+    {
+        if (audioConfig.numberOfChannels != kInvalidAudioChannels)
```

Comment:
Should we pass these values through even if invalid and let the server sort it out to keep client as thin as possible?

Reply:
ok, I will change the field type in IPC from optional to required for AudioConfig and SegmentAlignment and pass these values through always.